pkg/destroy/azure/azure.go

```diff
@@ -419,17 +423,63 @@ func deleteAzureStackPublicRecords(ctx context.Context, o *ClusterUninstaller) e
 	return utilerrors.NewAggregate(errs)
 }
 
-func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient, privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+func deleteRecordsFromBaseDomain(ctx context.Context, o *ClusterUninstaller) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	if o.BaseDomainResourceGroupName == "" || o.ZoneName == "" || o.APIURL == "" {
+		o.Logger.Debugf("could not find values in the metadata to get record set")
+		return nil
+	}
+
+	var errs []error
+	tag := fmt.Sprintf("kubernetes.io_cluster.%s", o.InfraID)
+	result, err := o.recordsClient.Get(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to get base domain dns zone: %w", err))
+		if isAuthError(err) {
+			return err
+		}
+	}
+
+	if value, ok := result.Metadata[tag]; ok {
+		if *value == "owned" {
```

Comment:
```suggestion
	if value, ok := result.Metadata[tag]; ok && *value == "owned"
```

Reply:
I added an else condition for a debug message.

Reply:
Ah I see it. Fixed.